src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -437,6 +438,7 @@ export class SQLFuture extends Disposable implements FutureInternal {
 			this.doneHandler.handle(msg);
 		}
 		this.doneDeferred.resolve(msg);
+		this._querySubsetResultMap.clear();
```

Comment:
Since this is the constructor, could we not just construct the Map here and skip clear()?